opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/hystrix/SleuthHystrixConcurrencyStrategy.java

```diff
@@ -0,0 +1,195 @@
+ * {@link Callable} that either starts a new span or continues one if the tracing was
+ * already running before the command was executed.
+ *
+ * @author Marcin Grzejszczak
+ * NOTE: Modified original by @author kameshsampath to suit OpenTracing
+ */
+public class SleuthHystrixConcurrencyStrategy extends HystrixConcurrencyStrategy {
+
+	private static final String HYSTRIX_COMPONENT = "hystrix";
+	private static final Log log = LogFactory
+			.getLog(SleuthHystrixConcurrencyStrategy.class);
+
+	private final Tracer tracer;
+	private HystrixConcurrencyStrategy delegate;
+
+	public SleuthHystrixConcurrencyStrategy(Tracer tracer) {
+		this.tracer = tracer;
+		try {
+			this.delegate = HystrixPlugins.getInstance().getConcurrencyStrategy();
+			if (this.delegate instanceof SleuthHystrixConcurrencyStrategy) {
+				// Welcome to singleton hell...
+				return;
+			}
+			HystrixCommandExecutionHook commandExecutionHook = HystrixPlugins
+					.getInstance().getCommandExecutionHook();
+			HystrixEventNotifier eventNotifier = HystrixPlugins.getInstance()
+					.getEventNotifier();
+			HystrixMetricsPublisher metricsPublisher = HystrixPlugins.getInstance()
+					.getMetricsPublisher();
+			HystrixPropertiesStrategy propertiesStrategy = HystrixPlugins.getInstance()
+					.getPropertiesStrategy();
+			logCurrentStateOfHysrixPlugins(eventNotifier, metricsPublisher,
+					propertiesStrategy);
+			HystrixPlugins.reset();
+			HystrixPlugins.getInstance().registerConcurrencyStrategy(this);
+			HystrixPlugins.getInstance()
+					.registerCommandExecutionHook(commandExecutionHook);
+			HystrixPlugins.getInstance().registerEventNotifier(eventNotifier);
+			HystrixPlugins.getInstance().registerMetricsPublisher(metricsPublisher);
+			HystrixPlugins.getInstance().registerPropertiesStrategy(propertiesStrategy);
+		}
+		catch (Exception e) {
+			log.error("Failed to register Sleuth Hystrix Concurrency Strategy", e);
+		}
+	}
+
+	private void logCurrentStateOfHysrixPlugins(HystrixEventNotifier eventNotifier,
+			HystrixMetricsPublisher metricsPublisher,
+			HystrixPropertiesStrategy propertiesStrategy) {
+		if (log.isDebugEnabled()) {
+			log.debug("Current Hystrix plugins configuration is [" + "concurrencyStrategy ["
+					+ this.delegate + "]," + "eventNotifier [" + eventNotifier + "],"
+					+ "metricPublisher [" + metricsPublisher + "]," + "propertiesStrategy ["
+					+ propertiesStrategy + "]," + "]");
+			log.debug("Registering Sleuth Hystrix Concurrency Strategy.");
+		}
+	}
+
+	@Override
+	public <T> Callable<T> wrapCallable(Callable<T> callable) {
+		//TODO: Cleanup Since TracedCallable does everything needed its not needed to wrap it in another callable
+//		if (callable instanceof HystrixTraceCallable) {
+//			return callable;
+//		}
+//		Callable<T> wrappedCallable = this.delegate != null
+//				? this.delegate.wrapCallable(callable) : callable;
+//		if (wrappedCallable instanceof HystrixTraceCallable) {
```

Comment:
Wondering whether we should have the same condition, could you please add it back with a comment when it happens?

Reply:
this related to next comment so i feel we can remove it